docs/source/howto/workflows.rst

```diff
@@ -135,171 +293,170 @@ The second argument is the result of the work chain, extracted from the ``Int``
-
-This is essentially the same as importing the workflow from its respective module, but using the ``WorkflowFactory`` has the advantage that the so called *entry point* (e.g. ``'arithmetic.multiply_add'``) will not change when the packages or plugins are reorganised.
-This means your code is less likely to break when updating AiiDA or the plugin that supplies the workflow.
-
-The list of installed plugins can be easily accessed via the verdi CLI:
-
-.. code-block:: console
-
-    $ verdi plugin list
-
-To see the list of workflow entry points, simply use:
+.. _how-to:workflows:extend:
 
-.. code-block:: console
+Extending workflows
+===================
 
-    $ verdi plugin list aiida.workflows
+When designing workflows, there are many cases where you want to reuse an existing process.
+This section explains how to extend workflows by wrapping them around other processes or linking them together.
 
-By further specifying the entry point of the workflow, you can see its description, inputs, outputs and exit codes:
+As an example, let's say you want to extend the ``MultiplyAddWorkChain`` by adding another step of analysis that checks whether the result is an even number or not.
+This final step can be written as a simple ``calcfunction``:
 
-.. code-block:: console
-
-    $ verdi plugin list aiida.workflows arithmetic.multiply_add
+.. literalinclude:: include/snippets/extend_workflows.py
+    :language: python
+    :pyobject: is_even
 
-Work functions
---------------
+We could simply write a new workflow based off ``MultiplyAddWorkChain`` that includes an extra step in the outline which runs the ``is_even`` calculation function.
+However, this would lead to a lot of code duplication, and longer workflows consisting of multiple work chains would become very cumbersome to deal with (see the dropdown panel below).
 
-Running a work function is as simple as calling a typical Python function: simply call it with the required input arguments:
+.. dropdown:: ``BadMultiplyAddIsEvenWorkChain``
 
-.. code-block:: python
+    .. literalinclude:: include/snippets/extend_workflows.py
+        :language: python
+        :pyobject: BadMultiplyAddIsEvenWorkChain
+.. note::
 
-    from aiida.plugins import WorkflowFactory, DataFactory
-    add_and_multiply = WorkflowFactory('arithmetic.add_multiply')
-    Int = DataFactory('int')
+    We've removed the ``result`` step from the outline, as well as the ``result`` output.
+    For this work chain, we're assuming that for now we are only interested in whether or not the result is even.
 
-    result = add_and_multiply(Int(2), Int(3), Int(5))
+We can avoid some code duplication by simply submitting the ``MultiplyAddWorkChain`` within one of the steps of our new work chain.
+In order to do this, we replace the ``multiply``, ``add`` and ``validate_result`` steps in the outline with one ``multiply_add`` step, in which we ``submit`` the ``MultiplyAddWorkChain`` and add the returned ``WorkChainNode`` to the context:
```

Comment:
If you are hiding code using the dropdown option, then I would say that the rest of the external explanation should not depend on this content. So, concretely, I would put the comments of the note above inside the dropdown and maybe describe the "good" workchain as a new thing instead of a modification of the "bad" one:

```suggestion
We can avoid some code duplication by simply submitting the ``MultiplyAddWorkChain`` within one of the steps of a new work chain which would then call `is_even` in a second step.
```